kahpp-spring-autoconfigure/src/main/java/dev/vox/platform/kahpp/actuator/KafkaStreamsState.java

```diff
@@ -0,0 +1,36 @@
+package dev.vox.platform.kahpp.actuator;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.streams.KafkaStreams;
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.kafka.config.StreamsBuilderFactoryBean;
+import org.springframework.stereotype.Component;
+
+@Component
+public class KafkaStreamsState extends AbstractHealthIndicator {
+
+  private final List<StreamsBuilderFactoryBean> streamsBuilders;
+
+  public KafkaStreamsState(List<StreamsBuilderFactoryBean> streamsBuilders) {
+    this.streamsBuilders = List.copyOf(streamsBuilders);
+  }
+
+  @Override
+  @SuppressWarnings("PMD.CloseResource")
+  protected void doHealthCheck(Health.Builder builder) {
+    Map<String, Status> streamsHealthMap = new HashMap<>();
+    for (StreamsBuilderFactoryBean stream : streamsBuilders) {
+      KafkaStreams kafkaStreams = stream.getKafkaStreams();
+      Status status =
+          kafkaStreams != null
```

Comment:
This nested ternary operator looks a bit difficult to read. If you want we can try to replace this block with java streams pipeline.

Reply:
Not sure if it is possible; how could you implement it?